### DISCUSSION TITLE
Zitadel JWT for Client assertion

### DISCUSSION THREAD

**Discussion by vikramsinghd** [zitadel/zitadel#7364] [Q&A]
Hi,
Is there any example for Create JWT for client assertion in .NET and then how we can use client assertion in our example. Please provide all steps for same.

**Comment by fforootd** [comment]
You could have a look into https://github.com/smartive/zitadel-net/blob/main/examples/Zitadel.ApiAccess/Program.cs that should have some answers for you.

**Reply by vikramsinghd** [reply]
Hi @fforootd, 
I have already gone through above example but their is services are missing and also i am not able to build the solution.

**Reply by fforootd** [reply]
And what is the error you see?

**Reply by vikramsinghd** [reply]
![Zitadel error](https://github.com/zitadel/zitadel/assets/157682045/82f9bddd-56ba-48d5-aa88-ff7a213f16db)

**Reply by vikramsinghd** [reply]
Hi @fforootd,

I am getting above error message.

**Reply by fforootd** [reply]
This looks oddly like you miss the dependencies of the client.

What version are you using of that package?

**Reply by vikramsinghd** [reply]
Hi @fforootd,
I am getting an error message for missing Metadata file for ZITADEL(net6.0 and net8.0) and also getting an error message for missing assembly references(**Admin,Auth,Management,AdminService,AdminService,ManagementService**). Do I need to create a service for same?

**Reply by fforootd** [reply]
I am no c# expert but should installing the nuget package do this ?

Have you installed this package https://github.com/smartive/zitadel-net/releases/tag/v6.1.2
